### PR TITLE
Add monkeytype (and retype and pytest-smartcov) 

### PR DESCRIPTION
**Pull request by bollwyvl**
Some more tools for observing and working with typed python.

- https://github.com/Instagram/MonkeyType
- https://github.com/ambv/retype
- also some testing stuff: https://github.com/carljm/pytest-smartcov

@nehaljwani, as the `mypy-feedstock` maintainer would you be interested in being added as a maintainer on any of these? Any feedback on dependency revision specs?

### CODE REVIEW

File: recipes/monkeytype/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [py2k]
+  skip: true  # [py34]
+  skip: true  # [py35]
```
Comment (nehaljwani) [inline recipes/monkeytype/meta.yaml]: This could probably just be:
```
skip: True  # [py<36]
```

File: recipes/retype/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [py2k]
+  skip: true  # [py34]
+  skip: true  # [py35]
```
Comment (nehaljwani) [inline recipes/retype/meta.yaml]: Same suggestion as above.

**Comment by nehaljwani** [comment]
@bollwyvl You could add me as maintainer, but I don't guarantee that I'll always keep them up to date. However, if there is PR, I'll tend to it within a day or two.

**Comment by bollwyvl** [comment]
@nehaljwani no worries, I'll put you on there, but I will certainly be adding them to my chores workflow! Thanks for the pointers on the selections!

**Comment by mwcraig** [comment]
Merging, thanks for address the license file in the source repos, @bollwyvl

**Comment by bollwyvl** [comment]
Yay!

**Pull request merged**